pkgs/by-name/do/dokieli/package.nix

```diff
@@ -1,69 +1,68 @@
 {
   lib,
-  stdenv,
   fetchFromGitHub,
-  fetchYarnDeps,
-  yarnConfigHook,
   makeWrapper,
-  yarnBuildHook,
-  yarnInstallHook,
-  nodejs,
+  nix-update-script,
+  nodePackages,
+  stdenv,
   xsel,
+  yarn-berry_4,
 }:
-
+let
+  yarn-berry = yarn-berry_4;
+in
 stdenv.mkDerivation (finalAttrs: {
   pname = "dokieli";
-  version = "0-unstable-2024-12-12";
+  version = "0-unstable-2025-07-06";
 
-  # Can't update newer versions currently because newer versions require yarn-berry, and it's not in nixpkgs, yet.
   src = fetchFromGitHub {
-    owner = "linkeddata";
+    owner = "dokieli";
     repo = "dokieli";
-    rev = "d8dc72c81b84ec12f791892a6377a7f6ec46ed3b";
-    hash = "sha256-CzSyQVyeJVOP8NCsa7ST3atG87V1KPSBzTRi0brMFYw=";
+    rev = "825d3aa5d754b512f8cbde2cc504b1d00788022c";
+    hash = "sha256-ZjsRG+qIzRb+8FM7a4qSkahxQeq1ayXSgD/oo7jdi6w=";
   };
 
-  offlineCache = fetchYarnDeps {
-    yarnLock = "${finalAttrs.src}/yarn.lock";
-    hash =
-      if stdenv.hostPlatform.isDarwin then
-        "sha256-bw5HszcHZ60qgYgm4qfhZEYXjJAQ2DXhWU0Reqb9VpQ="
-      else
-        "sha256-rwHBDBWZe4cdTyL7lNkB4nlpd5MWzbTU6kzdLBWcq0M=";
+  missingHashes = ./missing-hashes.json;
+  offlineCache = yarn-berry.fetchYarnBerryDeps {
+    inherit (finalAttrs) src missingHashes;
+    hash = "sha256-4SK1ecjEnnaow5Z2biCPaHirpX6J/5cytQWWicPgmB0=";
   };
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p $out/bin
     cp -r * $out
+
+    runHook postInstall
   '';
 
   nativeBuildInputs = [
     makeWrapper
-    yarnConfigHook
-    yarnBuildHook
-    yarnInstallHook
-    # Needed for executing package.json scripts
-    nodejs
+    yarn-berry.yarnBerryConfigHook
   ];
 
-  postFixup = ''
-    makeWrapper ${nodejs}/bin/npx $out/bin/dokieli           \
-      --prefix PATH : ${
-        lib.makeBinPath ([
-          nodejs
-          xsel
-        ])
-      }   \
-      --add-flags serve                                      \
-      --chdir $out/deps/dokieli
-  '';
+  postFixup =
+    let
+      serve = lib.getExe' nodePackages.serve "serve";
```

Comment:
`nodePackages` is deprecated: https://github.com/NixOS/nixpkgs/issues/229475
Please repackage serve if you want to keep using it.